src/Mollie/WC/Helper/PaymentMethodsIconUrl.php

```diff
@@ -39,7 +62,130 @@ protected function fallToAssets($paymentMethodName)
             return Mollie_WC_Plugin::getPluginUrl('public/images/' . $paymentMethodName . 's.svg');
         }
 
-        return Mollie_WC_Plugin::getPluginUrl('public/images/' . $paymentMethodName . '.svg');
+        return Mollie_WC_Plugin::getPluginUrl('public/images/' . $paymentMethodName . self::SVG_FILE_EXTENSION
+        );
+    }
+
+    /**
+     * Method that handles the composition of the new composed svg file
+     * and returns its url
+     *
+     * @param array $enabledCreditCards Array with the enabled creditcards
+     *
+     * @return string the url to the composed svg
+     */
+    protected function composeSvgImage($enabledCreditCards)
+    {
+        $this->writeToSvgFile();
+
+        $actual = $this->buildSvgComposed($enabledCreditCards);
+
+        return  $this->writeToSvgFile($actual);
+    }
```

Comment:
As far as I understanding you're calling the `writeToSvgFile` first time in order to create the file. Is that really necessary? Cannot you create a file directly with the right content?

Reply:
no, I call it to reset the content if there is any, but I guess I could do it all in one go.